stl/debugger/STL.natvis

```diff
@@ -1030,85 +1028,14 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-        <Size>_Mypair._Myval2._Mysize</Size>
-        <ValuePointer Condition="isShortString()">_Mypair._Myval2._Bx._Buf</ValuePointer>
-        <ValuePointer Condition="isLongString()">_Mypair._Myval2._Bx._Ptr</ValuePointer>
-      </ArrayItems>
-    </Expand>
-  </Type>
-
-  <Type Name="std::basic_string&lt;char32_t,*&gt;">
-    <Intrinsic Name="size" Expression="_Mypair._Myval2._Mysize" />
-    <Intrinsic Name="capacity" Expression="_Mypair._Myval2._Myres" />
-    <!-- _BUF_SIZE = 16 / sizeof(char32_t) &lt; 1 ? 1 : 16 / sizeof(char32_t) == 4 -->
-    <Intrinsic Name="bufSize" Expression="4" />
-    <Intrinsic Name="isShortString" Expression="capacity() &lt; bufSize()" />
-    <Intrinsic Name="isLongString" Expression="capacity() &gt;= bufSize()" />
-    <DisplayString Condition="isShortString()">{_Mypair._Myval2._Bx._Buf,s32}</DisplayString>
-    <DisplayString Condition="isLongString()">{_Mypair._Myval2._Bx._Ptr,s32}</DisplayString>
-    <StringView Condition="isShortString()">_Mypair._Myval2._Bx._Buf,s32</StringView>
-    <StringView Condition="isLongString()">_Mypair._Myval2._Bx._Ptr,s32</StringView>
-    <Expand>
-      <Item Name="[size]" ExcludeView="simple">size()</Item>
-      <Item Name="[capacity]" ExcludeView="simple">capacity()</Item>
-      <Item Name="[allocator]" ExcludeView="simple">_Mypair</Item>
-      <ArrayItems>
-        <Size>_Mypair._Myval2._Mysize</Size>
-        <ValuePointer Condition="isShortString()">_Mypair._Myval2._Bx._Buf</ValuePointer>
-        <ValuePointer Condition="isLongString()">_Mypair._Myval2._Bx._Ptr</ValuePointer>
-      </ArrayItems>
-    </Expand>
-  </Type>
-
-  <Type Name="std::_String_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;char&gt; &gt; &gt;">
-      <AlternativeType Name="std::_String_const_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;char&gt; &gt; &gt;" />
-      <AlternativeType Name="std::_String_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;char8_t&gt; &gt; &gt;" />
-      <AlternativeType Name="std::_String_const_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;char8_t&gt; &gt; &gt;" />
+  <Type Name="std::_String_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;*&gt; &gt; &gt;">
+      <AlternativeType Name="std::_String_const_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;*&gt; &gt; &gt;" />
       <SmartPointer Usage="Indexable">_Ptr,na</SmartPointer>
       <Expand>
           <Item Name="[ptr]">_Ptr</Item>
       </Expand>
   </Type>
 
-  <Type Name="std::_String_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;wchar_t&gt; &gt; &gt;">
-      <AlternativeType Name="std::_String_const_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;wchar_t&gt; &gt; &gt;" />
-      <AlternativeType Name="std::_String_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;unsigned short&gt; &gt; &gt;" />
-      <AlternativeType Name="std::_String_const_iterator&lt;std::_String_val&lt;std::_Simple_types&lt;unsigned short&gt; &gt; &gt;" />

```

Comment:
Ditto "restore `unsigned short` only"

Reply:
this whole PR is weird. Somehow `na` works for all "normal" string types, but not for `unsigned short`, which requires explicit `su`. Feels like it's some sort of bug in VS that made it work. Is it possible for you to check internally why `na` works for other types? Otherwise, with this PR, visuzlizers might be broken eventually if this whole thing worked due to internal VS bug

Reply:
`na` ("no address") should work for any pointer-to-object type. The problem is that `unsigned short` is not a character type, despite the existence of `/Zc:wchar_t-`, so `unsigned short*` doesn't trigger the text visualizer. We explicitly tell the debugger to print our `unsigned short*` like a `wchar_t*` with `su`. You could try `(char16_t*)meow,na` instead and see if that works well.

Reply:
> ... The problem is that `unsigned short` is not a character type, despite the existence of `/Zc:wchar_t-`, so `unsigned short*` doesn't trigger the text visualizer.

It's the opposite. Seems like text visualizer actually works with `unsigned short` with `na` when `/Zc:wchar_t-` is used. It doesn't work when `/Zc:wchar_t` is used. Looks like internally the debugger treats `unsigned short*` as `wchar_t*` when `/Zc:wchar_t-` is used.

So, overall, string/string_view visualizers could drop all overloads for `unsinged short` and simply use `na` everywhere. But that means that if some code uses `unsigned short` for strings, then it will start to show numbers

Reply:
> > ... The problem is that `unsigned short` is not a character type, despite the existence of `/Zc:wchar_t-`, so `unsigned short*` doesn't trigger the text visualizer.
> 
> It's the opposite. Seems like text visualizer actually works with `unsigned short` with `na` when `/Zc:wchar_t-` is used. It doesn't work when `/Zc:wchar_t` is used. Looks like internally the debugger treats `unsigned short*` as `wchar_t*` when `/Zc:wchar_t-` is used.
> 
> So, overall, string/string_view visualizers could drop all overloads for `unsinged short` and simply use `na` everywhere. But that means that if some code uses `unsigned short` for strings, then it will start to show numbers for regular builds without disabling native `wchar_t`.

Yes, you mentioned this in the other PR. I stand corrected - I probably shouldn't have conjectured about the behavior of `/Zc:wchar_t-` in the debugger when I haven't had to touch it for years. If it works with `/Zc:wchar_t-` then it's fine. We don't care in the slightest what happens to `basic_string<unsigned short>` without `/Zc:wchar_t-` because people shouldn't be using it in the first place. I retract both of these comments.

Reply:
I'll double check locally with string/string_view (both using only `na`) that all char types (`char`, `char8_t`, `char16_t`, `wchar_t`, `char32_t`) work identically regardless of `/Zc:wchar_t-` switch. In that case, string/string_view can be greatly simplified.

Reply:
Ah, great, we're on the same page now.

I saw this after my latest reply to the other PR. This slightly confused back-and-forth has me questioning the wisdom of my suggestion to split these changes into two PRs =)